frontend/packages/dev-console/src/components/AddPage.tsx

```diff
@@ -1,32 +1,62 @@
 import * as React from 'react';
 import { Helmet } from 'react-helmet';
+import * as _ from 'lodash';
 import { match as RMatch } from 'react-router';
+import { connect } from 'react-redux';
+import { RootState } from '@console/internal/redux';
+import { getActiveApplication } from '@console/internal/reducers/ui';
 import { history } from '@console/internal/components/utils';
 import { createProjectModal } from '@console/internal/components/modals';
+import { ALL_APPLICATIONS_KEY } from '@console/internal/const';
 import ODCEmptyState from './EmptyState';
 import NamespacedPage from './NamespacedPage';
 import DefaultPage from './DefaultPage';
+import TopologyDataController, { RenderProps } from './topology/TopologyDataController';
 
 export interface AddPageProps {
   match: RMatch<{
     ns?: string;
   }>;
 }
+interface StateProps {
+  activeApplication: string;
+}
+
+type props = AddPageProps & StateProps;
+
 const openProjectModal = () =>
   createProjectModal({
     blocking: true,
     onSubmit: (project) => history.push(`/add/ns/${project.metadata.name}`),
   });
-const AddPage: React.FC<AddPageProps> = ({ match }) => {
+
+const renderEmptyState = ({ data, loaded }: RenderProps) => {
+  return (
+    <ODCEmptyState
+      title="Add"
+      showHelpBlock={loaded ? _.isEmpty(data.graph.nodes) : false}
+      loaded={loaded}
+      hintBlockTitle="No workloads found"
+      hintBlockDescription="To add content to your project, create an application, component or service using one of these options."
+    />
+  );
+};
+
+const AddPage: React.FC<props> = ({ match, activeApplication }) => {
   const namespace = match.params.ns;
+  const application = activeApplication === ALL_APPLICATIONS_KEY ? undefined : activeApplication;
   return (
     <React.Fragment>
       <Helmet>
         <title>+Add</title>
       </Helmet>
       <NamespacedPage>
         {namespace ? (
-          <ODCEmptyState title="Add" />
+          <TopologyDataController
+            application={application}
+            namespace={namespace}
+            render={renderEmptyState}
+          />
```

Comment:
I am not sure if it makes sense to use `TopologyDataController` itself here, instead of this you can use the  utility function `getTopologyData` and from there you can get the `nodes`. Also, you wouldn't need `activeApplication` .

Reply:
Even If we use ```getTopologyData``` then we have to fetch all the resources and return the same data as ```TopologyController```.

Reply:
Why will you have to fetch all the resources? If you use `getTopologyData` it will return an object containing `graph` object which will have `nodes` array and then you can check its length and show the hint block accordingly. Am I missing something?

Reply:
Now, using ```Firehouse``` to fetch the workloads and check for empty or not.